sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryRequestHttpServletRequestProcessor.java

```diff
@@ -0,0 +1,57 @@
+package io.sentry.spring.boot;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.core.EventProcessor;
+import io.sentry.core.SentryEvent;
+import io.sentry.core.protocol.Request;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+/** Attaches information about HTTP request to {@link SentryEvent}. */
+@Open
+public class SentryRequestHttpServletRequestProcessor implements EventProcessor {
+
+  @Override
+  public @NotNull SentryEvent process(final @NotNull SentryEvent event, final @Nullable Object hint) {
+    final RequestAttributes requestAttributes = RequestContextHolder.currentRequestAttributes();
```

Comment:
We should generally make sure that our bugs don't affect the running app. That often means some not elegant code.

Here for example we could have this whole block on a try/catch and use the diagnostic logger to print out the error. That means we'd need to SentryOptions passed in (or the logger reference but the options allows setting up a logger after the SDK is initialized and swap the reference after that).

Alternatively we can catch exceptions of any event processor inside the SentryClient and log there. But some context would be gone I assume.

Reply:
It's not possible to inject `SentryOptions` here as it would create a circular dependency. I will add catching exceptions in `SentryClient` when processing events.